internal/app/ui/cpwsarea/wsmap/pmap/psettings/screenshot.go

```diff
@@ -57,12 +61,28 @@ func (p *Panel) showScreenshot() {
 
 func (p *Panel) createScreenshot() {
 	p.sessionScreenshot.saving = true
-
-	width, height := p.editor.Dmm().MaxX*dmmap.WorldIconSize, p.editor.Dmm().MaxY*dmmap.WorldIconSize
+	toolSelect := tools.SetSelected(tools.TNGrab).(*tools.ToolGrab)
```

Comment:
This will always switch user selected tool to `grab` when doing screenshot, which is undesirable.

1. Actually you don't need to get the grab tool if `inselection` mode is not enabled. You can check if the user set an `inselection` mode and then do the logic with bounds grabbing.
2. U need to get currently selected with `tools.Selected()` and check if it is the grab tool. Then do other logic.

Reply:
addressed i think

Reply:
addressed